airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -476,7 +476,7 @@ def __init__(
         number_logs_exception: int = 10,
         wait_for_completion: bool = True,
         waiter_delay: int = 6,
-        waiter_max_attempts: int = 100,
+        waiter_max_attempts: int = sys.maxsize, # Unless specified timeout is managed by airflow
```

Comment:
There are two things here:
1. Changing the default value to infinite could be a breaking change (cc: @eladkal )
2. If it's okay to change it, the best way is to make it None and replace the loop in the Trigger with `while True` when it's None.

Reply:
Agree it might be seen as breaking but looking at the problem description I think the previous change was accidentally breaking  (i.e. changing from effectively infinite to finite) and this one is simply a bugfix restoring it.

Reply:
Thanks for the review. I agree that adding `sys.maxsize` must have been a workaround and is a bit non-standard so perhaps this PR might be able to resolve that by using None.

I had considered if the `max_attempts` and `delay` parameters could be calculated at runtime by taking into consideration the `execution_timeout` value, however this would require a bit more of a rewrite of the Operator.

Reply:
> Agree it might be seen as breaking but looking at the problem description I think the previous change was accidentally breaking (i.e. changing from effectively infinite to finite) and this one is simply a bugfix restoring it.

Then I can accept this as non breaking change. Yet requesting to add a note explaining it in the top of the CHANGELOG. I will bake it to the proper release version during release process.

Reply:
Changelog note added [456ee50](https://github.com/apache/airflow/pull/33712/commits/456ee504884b21a94be52788501de1dba0e5bb02)

Unsure of the best way to refer back to the original PR that I am partially reverting, I trust you'll properly format the Changelog message when it comes to release time. TIA.